### ISSUE TITLE
bug(COMPONENT): IntelliSense is broken for v18 in standalone Components in VSCode

### ISSUE THREAD

**Issue by passee** [angular/angular#57847]
### Is this a regression?

- [X] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

17

### Description

@IliaPodolko
I hope its fine that I just copy the issue.
**This is a intentional duplicate!**

Installation of angular/material v18 breaks IntelliSense/Suggestions for some types from node_modules in VSCode. I tried the same repo steps using angular v17 and everything works fine.

**Important**
Angular language Service is up to date.

After including ""typeRoots": ["node_modules/@angular/material"]" in tsconfig under "compilerOptions" it works again, **BUT** you get an error in tsconfig:

> Cannot find type definition file for 'esm2022'.
  The file is in the program because:
    Entry point for implicit type library 'esm2022'ts
Cannot find type definition file for 'fesm2022'.
  The file is in the program because:
    Entry point for implicit type library 'fesm2022'ts
Cannot find type definition file for 'prebuilt-themes'.
  The file is in the program because:
    Entry point for implicit type library 'prebuilt-themes'ts
Cannot find type definition file for 'schematics'.
  The file is in the program because:
    Entry point for implicit type library 'schematics'ts

https://github.com/angular/components/issues/29537#issuecomment-2267167467
Is also not the solution. Its only a workaround. Intellisense should, like with the import in typeRoots, find the Module and import it on its own.

### Reproduction

Repo with this error https://github.com/IliaPodolko/vscode-imports
Steps to reproduce:

1. Create new angular app
    ng new my-angular-project
2. Install material
   npm install @angular/material
3. Go to app.component.ts and try add FormsModule, you will NOT see suggestions

### Expected Behavior

User should see suggestions in VSCode
![image](https://github.com/user-attachments/assets/f9879c34-9dea-4c78-af02-cadc40cf46c4)

### Actual Behavior

User does not see suggestions in VSCode when using angular v18 with angular/material

![image](https://github.com/user-attachments/assets/7ff72fb5-b156-4ff1-8ef0-ae04c0c10e0a)

### Environment

- Angular: 18.2.4 (CLI)
- CDK/Material: 18.2.4
- Browser(s): Chrome
- Operating System: Windows 11 64bit
- Node: 22.6.0

**Comment by JeanMeche** [comment]
Do you still have the issue if you restart the language service manually ? (You can do this via the VScode Command Palette).

**Comment by atscott** [comment]
Removing the language service tag. The Angular language service does not provide any autocomplete information at this location. It only provides Angular-specific information, mostly in templates, but sometimes in some other limited places (i.e. go to definition of `templateUrl` and `styleUrl`).

**Comment by passee** [comment]
> Do you still have the issue if you restart the language service manually ? (You can do this via the VScode Command Palette).

yes, but as @atscott said it should be not the problem. It still works/doesnt work when Angular language service (and all other extensions are disabled)

**Comment by atscott** [comment]
@passee Do you mean that enabling the Angular language service breaks this? That is, if you disable the Angular language service, it works?

**Comment by passee** [comment]
@atscott Sorry for the late response.
Nope, i mean regardless of if its activated it works and it does not. The decisive thing is whether or not the import in typeRoots is there.

**Comment by atscott** [comment]
Cc @crisbeto and @devversion again. As mentioned in the original issue, this doesn’t seem like a problem with Angular but rather VSCode or TypeScript.

**Comment by crisbeto** [comment]
Maybe it's worth reporting it to the TypeScript or vscode repositories? Our output format hasn't really changed in the past few years.

**Comment by devversion** [comment]
Agreed. Angular package format hasn't changed with regards to typings. This seems like a TypeScript language service issue at first glance.

**Comment by passee** [comment]
> It's likely you're hitting the 25-dependency cap (some packages can count for more than 1 under this heuristic); try setting   `includePackageJsonAutoImports`

 _Originally posted by @RyanCavanaugh in [#62301](https://github.com/microsoft/TypeScript/issues/62301#issuecomment-3230187802)_

it seems to be working with that typescript setting set to `on` in VSCode settings
